app/Http/Controllers/Plan/PlansController.php

```diff
@@ -609,40 +610,54 @@ public function completeDay(Request $request, $day_id)
         $user_is_member = $this->compareProjects($user->id, $this->key);
 
         if (!$user_is_member) {
-            return $this->setStatusCode(401)->replyWithError(trans('api.projects_users_not_connected'));
+            return $this
+                ->setStatusCode(SymfonyResponse::HTTP_UNAUTHORIZED)
+                ->replyWithError(trans('api.projects_users_not_connected'));
         }
 
         $plan_day = PlanDay::where('id', $day_id)->first();
 
         if (!$plan_day) {
-            return $this->setStatusCode(404)->replyWithError('Plan Day Not Found');
+            return $this
+                ->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)
+                ->replyWithError('Plan Day Not Found');
         }
 
-        $user_plan = UserPlan::join('plans', function ($join) use ($user) {
-            $join->on('user_plans.plan_id', '=', 'plans.id')->where('user_plans.user_id', $user->id);
-        })->where('user_plans.plan_id', $plan_day->plan_id)
-            ->select('user_plans.*')
-            ->first();
+        $user_plan = UserPlan::getByPlanIdAndUserId($plan_day->plan_id, $user->id);
 
         if (!$user_plan) {
-            return $this->setStatusCode(404)->replyWithError('User Plan Not Found');
+            return $this
+                ->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)
+                ->replyWithError('User Plan Not Found');
         }
 
         $complete = checkParam('complete') ?? true;
         $complete = $complete && $complete !== 'false';
 
-        $user_plan = \DB::transaction(function () use ($complete, $plan_day, $user, $user_plan) {
-            if ($complete) {
-                $plan_day->complete($user->id);
-            } else {
-                $plan_day->unComplete($user->id);
-            }
+        $result = null;
+        try {
+            \DB::transaction(function () use ($complete, $plan_day, $user) {
+                if ($complete) {
+                    $plan_day->complete($user->id);
+                } else {
+                    $plan_day->unComplete($user->id);
+                }
 
-            $user_plan->calculatePercentageCompleted()->save();
-            return $user_plan;
-        });
+                return $plan_day;
+            });
+            $result = $complete ? 'completed' : 'not completed';
+        } catch (\Exception $e) {
```

Comment:
don't catch all exceptions; just a SQL constraint violation. Specifically log that the exception was unique constraint violation.

Reply:
Hi @bradflood , I have updated it to catch specifically the SQL constraint violations as you can see in the following commit and line:

https://github.com/faithcomesbyhearing/dbp/pull/937/commits/50131cd2aaa5ca4a38819adf4bd5dbf8f8ecf668#diff-b2074cf830f20db0d05674376fd81b979af9a990752eb8e202431ca6729db2e4R864